matrix.py

```diff
@@ -0,0 +1,156 @@
+"""Matrix exercise, Daniel Prince - unity training."""
+
+
+class Matrix:
+    """Perform all sorts of matrix operations."""
+
+    def __init__(self, data):
+        self.data = data
+        # Check if n x n
+        for tup in self.data:
+            if len(self.data) != len(tup):
+                raise ValueError("not a valid matrix. (not nXn)")
+
+    def __str__(self):
+        return str(self.data)
+
+    def __repr__(self):
+        return '{self.__class__.__name__}({self.data})'.format(self=self)
+
+    def __iter__(self):
+        return iter(self.data)
+
+    def __hash__(self):
+        return hash(self.data)
+
+    def __eq__(self, other):
+        return self.data == other.data
+
+    def __ne__(self, other):
+        return self.data != other.data
+
+    def __add__(self, other):
+        """Adds 2 matrices."""
+
+        if isinstance(other, Matrix):
+            if len(self.data) == len(other.data):
```

Comment:
If you are using len(self.data) that much, consider saving it as a class property.

Reply:
That is good advice,  although now that its done, should I really go back and do it? its seems that it wouldn't be as officiant.
I would have to call the class property every time and it doesn't seem to save much time and effort considering It works. 
But like I said, I wish I would have thought of that before.
I hope to implement it to my next code.